src/main/java/org/cactoos/list/Solid.java

```diff
@@ -46,22 +44,16 @@ public Solid(final X... items) {
         this(new IterableOf<>(items));
     }
 
-    /**
-     * Ctor.
-     * @param items The array
-     */
-    public Solid(final Iterable<X> items) {
-        this(new CollectionOf<>(items));
-    }
-
     /**
      * Ctor.
      * @param list The iterable
      */
-    public Solid(final Collection<X> list) {
+    public Solid(final Iterable<X> list) {
         super(
-            new org.cactoos.scalar.Solid<>(
-                () -> new Synced<>(new Sticky<>(list))
+            new ListOf<>(
+                new org.cactoos.scalar.Solid<>(
+                    () -> new Synced<>(new Sticky<>(list))
```

Comment:
@victornoel change `new Synced<>(new Sticky<>(list))` by `new Solid<>(list)`

Reply:
@fabriciofx I'm currently implementing `Solid` here... I think this is out of scope of my job to change the way things are implemented btw: this PR is about changing some architectural aspects of cactoos collections, not adding or altering behaviour.